fetch_napari_data.py

```diff
@@ -292,68 +263,125 @@ def build_plugins_dataframe() -> pd.DataFrame:
-        pypi_info = fetch(urljoin(API_PYPI_BASE_URL, plugin_normalized_name))
-
-        expand_proj_url(plugin_data)
-
-        if conda_info:
-            # we only want a limited set of conda info
-            conda_info = {
-                "conda_name": conda_info["name"],
-                "conda_html_url": conda_info["html_url"],
-            }
-            plugin_data.update(conda_info)
-
-        if pypi_info:
-            # releases are sorted in descending order
-            plugin_first_release = plugin_data["pypi_versions"][-1]
-            plugin_latest_release = plugin_data["pypi_versions"][0]
-            initial_release_date = get_version_release_date(
-                pypi_info, plugin_first_release
-            )
-            last_updated_date = get_version_release_date(
-                pypi_info, plugin_latest_release
-            )
-
-            # grab pypi project link
-            home_pypi = pypi_info["info"].get("package_url", "")
-            plugin_data.update(
-                {
-                    "created_at": initial_release_date,
-                    "modified_at": last_updated_date,
-                    "home_pypi": home_pypi,
-                }
-            )
-
-        if manifest_info:
-            flatten_and_merge(plugin_data, manifest_info)
-
-        all_plugin_data.append(plugin_data)
-
-    with ThreadPoolExecutor() as executor:
-        executor.map(process_plugin, summary_df)
-
-    return pd.DataFrame(all_plugin_data)
+    )
+
+    widgets = [widget["display_name"] for widget in contributions["widgets"] or []]
+
+    sample_data = [
+        sample["display_name"] for sample in contributions["sample_data"] or []
+    ]
+
+    return PluginPageData(
+        normalized_name=plugin_normalized_name,
+        name=name,
+        display_name=display_name,
+        version=plugin_latest_release,
+        created_at=initial_release_date,
+        modified_at=last_updated_date,
+        authors=authors,
+        author_emails=emails,
+        license=package_license,
+        home_pypi=home_pypi,
+        home_github=home_github,
+        home_other=home_other,
+        summary=plugin_summary_data.get("summary", ""),
+        package_metadata_requires_python=package_metadata.get("requires_python", ""),
+        package_metadata_requires_dist=package_metadata.get("requires_dist", []),
+        package_metadata_description=package_metadata.get("description", ""),
+        package_metadata_classifiers=package_metadata.get("classifiers", []),
+        contributions_readers_filename_patterns=reader_patterns,
+        contributions_writers_filename_extensions=writer_extensions,
+        contributions_widgets=widgets,
+        contributions_sample_data=sample_data,
+    )
+
+
+def create_search_index(plugins: list[PluginPageData], data_dir: str):
+    search_index = [
+        {
+            "name": plugin.name,
+            "display_name": plugin.display_name,
```

Comment:
Would be good to include the manifest `categories` field here too. I don't think it's that commonly used, but if it is used, it's likely to be useful for search.

Reply:
I added it to the search index, though I'm realizing it's not even shown on the plugin page!

I looked at adding it to the template but...we desperately need #49/#75 haha.